lib/msf/util/exe.rb

```diff
@@ -671,6 +671,31 @@ def self.to_win64pe_service(framework, code, opts = {})
     exe_sub_method(code,opts)
   end
 
+  # self.set_template_default_winpe_dll
+  #
+  # Set the default winpe DLL template. It will select the template based on the parameters provided including the size
+  # architecture and an optional flavor. See data/templates/src/pe for template source code and build tools.
+  #
+  # @param opts [Hash]
+  # @param arch The architecture, as one the predefined constants.
+  # @param size [Integer] The size of the payload.
+  # @param flavor [Nil,String] An optional DLL flavor, one of 'mixed_mode' or 'dccw_gdiplus'
+  private_class_method def self.set_template_default_winpe_dll(opts, arch, size, flavor: nil)
+    return if opts[:template].present?
+
+    # dynamic size upgrading is only available when MSF selects the template because there's currently no way to
+    # determine the amount of space that is available in the template provided by the user so it's assumed to be 4KiB
+    match = {4096 => '', 262144 => '.256kib'}.find { |k,v| size < k }
```

Comment:
```suggestion
    match = {4096 => '', 262144 => '.256kib'}.find { |k,v| size <= k }
```

Think this should be less than or equal to since otherwise we will have an off by one condition here. This assumes the size copied is the same as `size` and we don't have something like a extra null terminating byte that will be copied though.